stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/internal/MessageListItemDecoratorProvider.kt

```diff
@@ -42,16 +43,17 @@ import io.getstream.chat.android.ui.message.list.background.MessageBackgroundFac
  * @param messageListViewStyle [MessageListViewStyle] The style of the MessageListView and its items.
  * @param showAvatarPredicate [MessageListView.ShowAvatarPredicate] Checks if should show the avatar or not accordingly with the provided logic.
  * @param messageBackgroundFactory [MessageBackgroundFactory] Factory that customizes the background of messages.
- * @param deletedMessageListItemPredicate [MessageListView.MessageListItemPredicate] Predicate to hide or show the the deleted message accordingly to the logic provided.
+ * @param deletedMessageVisibility [DeletedMessageVisibility] Used to hide or show the the deleted message accordingly to the logic provided.
  * @param isCurrentUserBanned Checks if the current user is banned inside the channel. Used for failed icon indicator.
  */
+@Suppress("LongParameterList")
 internal class MessageListItemDecoratorProvider(
     dateFormatter: DateFormatter,
     isDirectMessage: () -> Boolean,
     messageListViewStyle: MessageListViewStyle,
     showAvatarPredicate: MessageListView.ShowAvatarPredicate,
     messageBackgroundFactory: MessageBackgroundFactory,
-    deletedMessageListItemPredicate: MessageListView.MessageListItemPredicate,
```

Comment:
This part is problematic because it is initialized once and does not respond to updates.
It means that once the view is initialized, the footer will display the essage with regards to the deleted messages visibility mode that was set before initialization.

I am unsure how to communicate this to the users.
Throwing if the user sets this after initialization might be weird for SDK users because they would be setting the visibility on the `MessageListViewModel` but receiving an exception from the `MessageListView`.

Reply:
what about throwing an exception in `MessageListView:: setDeletedMessageVisibility` we can check if adapter was already initialized